File: server/src/main/java/gmetrica/enumeration/MetodoContagem.java
package gmetrica.enumeration;

public enum MetodoContagem {
	
	CONTAGEM_DETALHADA ("Contagem Detalhada"),
	CONTAGEM_ESTIMADA_NESMA ("Contagem Estimada NESMA"),
	CONTAGEM_ESTIMATIVA ("Contagem Estimativa");
	
	private String descricao;
	
	private MetodoContagem(String descricao) {
		this.descricao = descricao;
	}

	public String getDescricao() {
		return descricao;
	}

	public static MetodoContagem fromDescricao(String descricao) {
		if(descricao == null) {
			return null;
		}
		
		for(MetodoContagem valor: MetodoContagem.values()) {
			
			if(valor.getDescricao().equals(descricao)) {
				return valor;
			}
		}
		
		return null;
	}
	
	public static MetodoContagem fromContains(String descricao) {
		if(descricao == null) {
			return null;
		}
		
		for(MetodoContagem valor: MetodoContagem.values()) {
			descricao = descricao.substring(0,1).toUpperCase().concat(descricao.substring(1));
			if(valor.getDescricao().contains(descricao)) {
				return valor;
			}
		}
		
		return null;
	}

}

File: client/src/app/enumeration/metodo-contagem.ts
import { SelectItem } from "primeng/components/common/selectitem";

export enum MetodoContagem {
    CONTAGEM_DETALHADA = <any>"Contagem Detalhada",
    CONTAGEM_ESTIMADA_NESMA = <any>"Contagem Estimada NESMA",
    CONTAGEM_ESTIMATIVA = <any>"Contagem Estimativa"
}

export namespace MetodoContagem {
    export function getSelectItem(): any {
        var item: SelectItem[] = [];
        item.push({label:'Selecione', value:null});
        item.push({label:MetodoContagem.CONTAGEM_DETALHADA.toString(), value:MetodoContagem[MetodoContagem.CONTAGEM_DETALHADA]});
        item.push({label:MetodoContagem.CONTAGEM_ESTIMADA_NESMA.toString(), value:MetodoContagem[MetodoContagem.CONTAGEM_ESTIMADA_NESMA]});
        item.push({label:MetodoContagem.CONTAGEM_ESTIMATIVA.toString(), value:MetodoContagem[MetodoContagem.CONTAGEM_ESTIMATIVA]});
        return item;
    } 
}